test/puppetlabs/trapperkeeper/services/status/status_service_test.clj

```diff
@@ -176,3 +176,46 @@
                 "message" (str "No status function with version 3 "
                             "found for service foo")}
               (json/parse-string (slurp (:body resp)))))))))
+
+(defn response->status
+  [resp]
+  (:status (json/parse-string (slurp (:body resp)) true)))
+
+(deftest level>=test
+  (testing "use of level>= to implement a status function"
+    (let [my-status (fn [level]
+                      (let [level>= (partial status-core/level>= level)]
+                        {:is-running true
+                         :status (cond-> {:this-is-critical "foo"}
+                                   (level>= :info) (assoc :bar "bar"
+                                                          :baz "baz")
+                                   (level>= :debug) (assoc :x "x"
+                                                           :y "y"
+                                                           :z "y"))}))
```

Comment:
I think you're on to something there ... abstracting-out the comparison seems like a good idea.  However, I like the convenience of having a pre-defined `level>=` function - and it seems like similar functions like `level<=` and `level=` functions would also be useful - and all 3 of these could be implemented on top of something like ...

```clj
(defn compare-levels
  [operator level1 level2]
  (operator (level->numeric level1) (level->numeric level2))
```

.. whacha think of that?

Switching topics to one of the 2 Hard Problems of Computer Science ...

Sure, I'm not married to `level->numeric`, although it seems a bit clearer than `level-value` ... how about `numeric-value` or `numeric-level-value`?